### ISSUE TITLE
Questions about the debanding profiles and hdeband shaders

### ISSUE THREAD

**Issue by Zabooby** [dyphire/mpv-config#31]
For your three debanding profiles you only change the number of iterations and I had a few suggestions/questions to improve them. 

- Shouldn't the grain be lower for the low/medium profiles as the number of artefacts wouldn't be as noticeable? 
- As the manual states, shouldn't the range decrease with each increase in the number of iterations?
- Why is the threshold so high for the low/medium profiles, shouldn't it be lower considering that higher numbers reduce image details, also considering someone would use these profiles for low levels of banding anyway? 

TLDR: The profiles are not as tuned as they should be. 

I came across [this](https://github.com/hooke007/MPV_lazy/discussions/336#discussioncomment-6456197) discussion about the hdeband shaders a while ago and brought it up with @AN3223, who has since made some updates to the shaders. Have you tested the latest versions yet, if so what are your thoughts?

**Comment by dyphire** [comment]
> For your three debanding profiles you only change the number of iterations and I had a few suggestions/questions to improve them.
> 
> * Shouldn't the grain be lower for the low/medium profiles as the number of artefacts wouldn't be as noticeable?
> * As the manual states, shouldn't the range decrease with each increase in the number of iterations?
> * Why is the threshold so high for the low/medium profiles, shouldn't it be lower considering that higher numbers reduce image details, also considering someone would use these profiles for low levels of banding anyway?
> 
> TLDR: The profiles are not as tuned as they should be.

Yes, there are some issues with the three profiles of Deband, which are legacy issues. 
In fact, using the mpv preset deband parameter is sufficient for handling weak bands. The profile of DeBand-medium can handle medium to high strength bands well. And I think both of mpv's dband and  hdeband.glsl are powerless for high-strength and extremely poor ribbons. Therefore, these two profiles of DeBand-low and DeBand-high are actually a bit redundant. 
As for my preset profile is DeBand-medium, it is only a personal habit, and I recommend using the mpv preset deband parameter by default.

> I came across [this](https://github.com/hooke007/MPV_lazy/discussions/336#discussioncomment-6456197) discussion about the hdeband shaders a while ago and brought it up with @AN3223, who has since made some updates to the shaders. Have you tested the latest versions yet, if so what are your thoughts?

I am not interested in the shader of hdeband.glsl as it has no significant advantage over the internal deband of mpv and only works on `--vo=gpu-next`.
I don't object to using it, but I won't recommend it

**Comment by Zabooby** [comment]
Most of the time, I watch videos with little to no banding so should I just stick to mpv's default parameters and for any higher levels of debanding use your medium profile?

**Comment by dyphire** [comment]
> Most of the time, I watch videos with little to no banding so should I just stick to mpv's default parameters and for any higher levels of debanding use your medium profile?

Yep

**Comment by AN3223** [comment]
> I am not interested in the shader of hdeband.glsl as it has no significant advantage over the internal deband of mpv and only works on `--vo=gpu-next`. I don't object to using it, but I won't recommend it.

I just pushed some commits, try it out and let me know what you think :)

Also it works fine on `--vo=gpu`  for me, is this still an issue for you?
